src/encoder/encodersndfileflac.cpp

```diff
@@ -1,11 +1,40 @@
+#include "encoder/encodersndfileflac.h"
+
 #include <QtDebug>
 
-#include "encoder/encodersndfileflac.h"
 #include "encoder/encodercallback.h"
+#include "util/sample.h"
+
+namespace {
+constexpr SINT kEncBufferSize = 8192; // used inside libsndfile for flac
+
+//static
+void convertFloat32ToIntFormat(int* pDest, const CSAMPLE* pSrc, SINT numSamples, int format) {
+    const CSAMPLE kConversionFactor = INT_MIN * -1.0f;
+    if (format & SF_FORMAT_PCM_16) {
+        // note: LOOP VECTORIZED only with "int i"
```

Comment:
Unrelated: The knowledge and implicit assumptions about vectorization are scattered. One more reason define and document a common typedef in sample.h.

Reply:
I have tested and SINT works here. I have originally just blindly copied the original implementation including the comment.